Modules/Loadable/SubjectHierarchy/Widgets/qMRMLSubjectHierarchyTreeView.cxx

```diff
@@ -603,6 +606,37 @@ void qMRMLSubjectHierarchyTreeView::updateRootItem()
 }
 
 //--------------------------------------------------------------------------
+void qMRMLSubjectHierarchyTreeView::updateItemsSelectableFlag()
+{
+  Q_D(qMRMLSubjectHierarchyTreeView);
+  if (!d->SubjectHierarchyNode)
+    {
+    return;
+    }
+
+  qMRMLSubjectHierarchyModel* sceneModel = qobject_cast<qMRMLSubjectHierarchyModel*>(this->model());
+
+  // Collect selected subject hierarchy items
+  std::vector<vtkIdType> childItemIDs;
+  vtkIdType sceneId = d->SubjectHierarchyNode->GetSceneItemID();
+  d->SubjectHierarchyNode->GetItemChildren(sceneId, childItemIDs, true);
+  foreach(vtkIdType itemID, childItemIDs)
+    {
+    bool isSelectable = this->sortFilterProxyModel()->filterAcceptsItem(itemID, false);
+    QStandardItem* item = sceneModel->itemFromSubjectHierarchyItem(itemID); // or proxy model?
```

Comment:
- it seems that the `QStandardItems` are shared between treeviews,
so changing their selection impacts all the other treeviews
- is it ok to use `itemFromSubjectHierarchyItem` from the scenemodel?
Should there be a mapping with the sortFilterProxyModel? The latter
however does not have a `itemFromSubjectHierarchyItem` method.

Reply:
* Cannot be possible. QStandardItems are the view items that are displayed in each treeview separately. Please read http://doc.qt.io/qt-4.8/model-view-programming.html and http://doc.qt.io/qt-4.8/modelview.html
* Using that function from the scene model when there is a proxy filter will probably make the application crash. Use the proxy model's converters instead.

Reply:
What is strange that the tree views create their own model and proxymodel in qMRMLSubjectHierarchyTreeViewPrivate::init. I don't see how it is possible that the filters are applied to all of the views. Also I cannot reproduce this behavior, the filters are correctly applied on the separate widgets individually

Reply:
> Use the proxy model's converters instead.

What do you mean by converter here? Are you saying I should implement `itemFromSubjectHierarchyItem` in the proxy model class, and properly use the mapping within it?

Reply:
If it's not there yet then yes please. You only need to call the same mapping function as the others and call the scene model's converter.